### PR TITLE
GoCardless capture

### PR DESCRIPTION
**Pull request by steve-billforward**
Adds the ability to capture a bank account using GoCardless.

Still to do:
- update readme
- add example HTML file

### CODE REVIEW

File: src/billforward.js [hunk 1]
```diff
@@ -2947,6 +2972,73 @@
...
+            return function() {
+                return new lambda(arguments);
+            }
+        })();
+
+        p['do'] = function() {
+            // do nothing
+        };
+
+        p.doSubmitDanceWhenReady = function () {
```
Comment (mjaggard) [inline src/billforward.js:3003]: Is the term `dance` used elsewhere in this repo? If not, maybe it should be more official-looking?

File: src/billforward.js [hunk 2]
```diff
@@ -3249,9 +3343,9 @@
     };
 
     bfjs.captureBankAccount = function(bankAccountDetails, targetGateway, accountID, callback) {
-        if (targetGateway != "stripe") {
+        if (!['stripe', 'goCardless', 'gocardless'].includes(targetGateway)) {
             throw "'" + targetGateway + "' gateway does not support bank accounts";
-        } else {
+        } else if (targetGateway === 'stripe') {
```
Comment (mjaggard) [inline src/billforward.js]: Would this if/else statement make more sense in a different order:
`if (stripe) ... else if (gocardless) ... else ....` ?

File: src/billforward.js (hunk 1 above)
Comment (steve-billforward) [reply src/billforward.js:3003]: Yeah, this function name is also used for the other payment gateways. I'm not convinced it's a good name, but I wanted to be consistent.

File: src/billforward.js (hunk 2 above)
Comment (steve-billforward) [reply src/billforward.js]: Hmm, I think all I'm going to do is separate them into two independent `if` statements. The first is a validation step ("is this a valid gateway?") and the second is some kind of trick with the Stripe gateway that I don't understand.

So
```js
if (!['stripe', 'goCardless', 'gocardless'].includes(targetGateway)) {
    throw "'" + targetGateway + "' gateway does not support bank accounts";
}

if (targetGateway === 'stripe') {
    // We are masking the stripe ach gateway as a plain stripe one
    targetGateway = "stripeach";
}
```

**Review by mjaggard — APPROVED** [review]

**Pull request merged**